src/test/unit_tests/test_buffer_pool.js

```diff
@@ -44,4 +44,53 @@ mocha.describe('Test buffers pool', function() {
         assert(buffers_pool.sem.value === SEM_LIMIT, 'Smepahore did not deallocate after buffers release');
     });
 
+    mocha.it('Work parallel with buffers with different sizes and different semaphores', async function() {
```

Comment:
add test case for undefined and negative sizes passed to get_buffer_pool that return the largest, but 0 should return the smallest.